src/sna/toots-sna.ipynb

```diff
@@ -268,16 +214,234 @@
+    "        df_bet_cen.reset_index(level=0, inplace=True)\n",
+    "        analyse= pd.merge(analyse, df_bet_cen, on = ['ID'])\n",
+    "\n",
+    "    # Clustering coefficient \n",
+    "    clust_coefficients = nx.clustering(G)\n",
+    "    df_clust = pd.DataFrame.from_dict(clust_coefficients, orient='index')\n",
+    "    df_clust.columns = ['clust_coefficient']\n",
+    "    df_clust.index.names = ['ID']\n",
+    "    df_clust.reset_index(level=0, inplace=True)\n",
+    "    analyse= pd.merge(analyse, df_clust, on = ['ID'])\n",
+    "\n",
+    "    # Closeness centrality \n",
+    "    if not fast:\n",
+    "        clo_cen = nx.closeness_centrality(G)\n",
+    "        df_clo = pd.DataFrame.from_dict(clo_cen, orient='index')\n",
+    "        df_clo.columns = ['closeness_centrality']\n",
+    "        df_clo.index.names = ['ID']\n",
+    "        df_clo.reset_index(level=0, inplace=True)\n",
+    "        analyse= pd.merge(analyse, df_clo, on = ['ID'])\n",
+    "\n",
+    "    # Eigenvector centrality\n",
+    "    #eig_cen = nx.eigenvector_centrality(G)\n",
+    "    #eig_cen = nx.katz_centrality(G)\n",
+    "    eig_cen = nx.eigenvector_centrality_numpy(G)\n",
+    "    df_eig = pd.DataFrame.from_dict(eig_cen, orient='index')\n",
+    "    df_eig.columns = ['eigenvector_centrality']\n",
+    "    df_eig.index.names = ['ID']\n",
+    "    df_eig.reset_index(level=0, inplace=True)\n",
+    "    analyse= pd.merge(analyse, df_eig, on = ['ID'])\n",
+    "    return analyse"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2164b25f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Sort betweeness and degree.  A node can have a lot of connection, but influence various by WHO you are connected to.\n",
+    "# Downloaded centrality_15K.parquet from se\n",
+    "should_regenerate = False\n",
+    "if should_regenerate:\n",
+    "    status_gph = create_graph_from_status(statuses_df, False)\n",
+    "    centrality = analyze_centrality_1(status_gph)\n",
+    "    centrality.to_parquet(\"{}/results/centrality_15K.parquet\".format(data_path))\n",
+    "else:\n",
+    "    centrality = pd.read_parquet(\"{}/results/centrality_15K.parquet\".format(data_path))\n",
+    "\n",
+    "print('Total Results {}'.format(len(status_gph)))\n",
+    "median_degree = centrality['degree']\n",
+    "print('Top 25 Sorted by Betweeness')\n",
+    "display(centrality.sort_values('betweenness_centrality', ascending=False).head(25))\n",
+    "\n",
+    "print('Top 25 Sorted by  Degree')\n",
+    "display(centrality.sort_values('degree', ascending=False).head(50))\n",
+    "\n",
+    "\n",
+    "# Note to lookup ID - we need to use the `/api/v1/accounts` and look up the URL"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "862d8e85",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_graph_from_user(user_id, degrees_to_search = 2):\n",
```

Comment:
could you pls add some comments here ?
is this for getting a subgraph involving the user_id ?